Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1492,12 +1526,32 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes()
         }
     };
 
-    auto computeIndefiniteItems = [&](SingleThreadWeakListHashSet<RenderBox>& indefiniteItems) {
+    auto computeIndefiniteItems = [&](SingleThreadWeakListHashSet<RenderBox>& singleTrackMasonryIndefiniteItems, SingleThreadWeakListHashSet<RenderBox>& masonryIndefiniteItems, LayoutUnit& singleTrackMinContentForChild, LayoutUnit& singleTrackMaxContentForChild, LayoutUnit &singleTrackMinSizeForChild) {
         for (auto trackIndex : m_contentSizedTracksIndex) {
             GridIterator iterator(m_grid, m_direction, trackIndex);
             while (CheckedPtr<RenderBox> gridItem = iterator.nextGridItem()) {
-                if (GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, *gridItem, m_direction).isIndefinite())
-                    indefiniteItems.add(*gridItem);
+                if (GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, *gridItem, m_direction).isIndefinite()) {
+                    auto spanLength = m_renderGrid->gridSpanForChild(*gridItem, m_direction).integerSpan();
+
+                    if (spanLength == 1 && !gridItem->isRenderGrid()) {
```

Comment:
Is there a reason why we do not allow grid items which are grids? In the comment you mention we do not include subgrids in this optimization but this by itself will also exclude grids which are grid items. If we want to exclude subgrids here we will want to expand on this condition a bit